cob_scan_unifier/src/scan_unifier_node.cpp

```diff
@@ -212,7 +212,7 @@ bool ScanUnifierNode::unifyLaserScans(std::vector<sensor_msgs::LaserScan::ConstP
       try
       {
         if (!listener_.waitForTransform(frame_, current_scans.at(i)->header.frame_id,
-                                        current_scans.at(i)->header.stamp, ros::Duration(1.0)))
+                                        current_scans.at(i)->header.stamp + ros::Duration().fromSec((current_scans.at(i)->ranges.size()-1)*current_scans.at(i)->time_increment), ros::Duration(1.0)))
```

Comment:
why did you use `ranges.size()-1`, instead of `ranges.size()`?
your source states that you need to wait for the last scan, i.e. wait `time_increment` for every range value...with `-1` you are missing one `time_increment`...

Reply:
Alright changed as requested.
```suggestion
                                        current_scans.at(i)->header.stamp + ros::Duration().fromSec(current_scans.at(i)->ranges.size()*current_scans.at(i)->time_increment), ros::Duration(1.0)))
```